### ISSUE TITLE
Incorrect reporting of Indentation issues for IntelliJ highlighting

### ISSUE THREAD

**Issue by godiale** [detekt/detekt#3342]
## Expected Behavior
Intellij IDEA highlights a line (or first symbol in a line), where **Indentation** issue is found.
Hoovering mouse over highlighted area shows tooltip with the issue description.

Below is screenshot with correct highlighting and issue description from **SpacingAroundParens** rule.
![correct_spacing_highlighting](https://user-images.githubusercontent.com/11892279/103480458-9524d280-4de5-11eb-8e18-23fbed15603a.png)

## Observed Behavior
Whole file content is highlighted, making it impossible to find violation (including other types of rules).
Hoovering mouse over random place of file shows correct violation description, but its impossible to find its location.

Below is screenshot with the problem from testing project to reproduce the issue.
![incorrect_indentation_highlighting](https://user-images.githubusercontent.com/11892279/103480466-9fdf6780-4de5-11eb-95b3-868ebe67d1b9.png)

## Steps to Reproduce
- Download test project: https://github.com/godiale/detekt-indentation-highlighting-issue
- Build it with ```./gradlew.bat build```, which will result in following output
```
1 kotlin file was analyzed.
formatting - 10min debt
        Indentation - [] at C:\Users\godin\Projects\detekt-indentation-highlighting-issue\src\main\kotlin\MainWithFormattingErrors.kt:4:1
        SpacingAroundParens - [] at C:\Users\godin\Projects\detekt-indentation-highlighting-issue\src\main\kotlin\MainWithFormattingErrors.kt:5:20

Overall debt: 10min
```
- Install detekt IntelliJ plugin and configure it as below:
    - Enable Detekt = Y
    - Enable Formatting rules = Y
    - Build upon the default configuration = N
    - Enable all experimental rules = N
    - Treat detekt findings as errors = N
    - Configuration file = https://github.com/godiale/detekt-indentation-highlighting-issue/blob/main/config/detekt.yml
    - Baseline File = <empty>
    - Plugin jars = <empty>
![idea_settings](https://user-images.githubusercontent.com/11892279/103480471-a837a280-4de5-11eb-914a-1a47f34bcc36.png)
- Open ```MainWithFormattingErrors.kt``` in editor

## Context
The issue duplicates [detekt-intellij-plugin#105](https://github.com/detekt/detekt-intellij-plugin/issues/105), with more reproduction details.
However I think placing it to detekt project is more correct, because seems the problem is not in visualizing code smell, but rather in the way how its reported.
I built detekt itself to investigate and found that ```location``` in [**FormattingRule**](https://github.com/detekt/detekt/blob/v1.15.0/detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/FormattingRule.kt#L65) for Indentation is created with values like ```TextLocation(0, 1165)```where ```node.psi.endOffset``` is equal to index of last character in source file.
```kotlin
            val location = Location(
                SourceLocation(line, column),
                TextLocation(node.startOffset, node.psi.endOffset),
                root.absolutePath().toString()
            )
```
```SourceLocation``` is created with correct line and column is reported as 1.
Maybe this could be fixed by customizing ```TextLocation``` creation for ```Indentation```, so that only affected line is put to ```TextLocation```, not whole file?

## Your Environment
* Version of detekt used: 1.15.0
* Version of detekt IntelliJ plugin used: 1.6.1
* Version of Gradle used (if applicable): 6.7
* Operating System and version: Windows 10 Pro
* Version of IntelliJ IDEA: Community 2020.3.1 Build #IC-203.6682.168, built on December 29, 2020
* Runtime version: 11.0.9.1+11-b1145.63 amd64
* Link to your project (if it's a public repository): https://github.com/godiale/detekt-indentation-highlighting-issue

**Comment by flapjackfritz** [comment]
This doesn't appear to be an issue with detekt, it seems to just be an issue with the plugin.

When I run using the detekt cli,
e.g. `detekt -c default-detekt-config.yml -i ./src/main/kotlin/nz/govt/linz/landonline/survey/controller/MyFirmController.kt --plugins ../detekt-formatting-1.17.1.jar`
I get result:
```
1 kotlin file was analyzed.
/Users/tommyfritz/workspace/linz/landonline-survey/src/main/kotlin/nz/govt/linz/landonline/survey/controller/MyFirmController.kt - 10min debt
	Indentation - [] at /Users/tommyfritz/workspace/linz/landonline-survey/src/main/kotlin/nz/govt/linz/landonline/survey/controller/MyFirmController.kt:29:34
	Indentation - [] at /Users/tommyfritz/workspace/linz/landonline-survey/src/main/kotlin/nz/govt/linz/landonline/survey/controller/MyFirmController.kt:36:9

Overall debt: 10min
```

Which is indicating the correct line numbers for this issue (line 29 and 36). So Detekt is working.

But in the plugin, it seems to fail to figure out these lines, instead saying it's on line 1:
<img width="879" alt="Screen Shot 2021-06-18 at 1 14 09 PM" src="https://user-images.githubusercontent.com/1535301/122491000-438c9f00-d037-11eb-97e1-f02848285e54.png">

So I'm not sure what's going on here. I've looked through the plugin code, and I can't figure out where it's going wrong.

**Comment by cortinico** [comment]
> This doesn't appear to be an issue with detekt, it seems to just be an issue with the plugin.

I believe we should move this to the detekt plugin issue board.
@schalkms I saw you closed #105 pointing to this issue instead. If this is a plugin issue, I'd rather keep the discussion on the other repo.

**Comment by schalkms** [comment]
@cortinico that's fine for me.
